### PR TITLE
Breaking Docs for Bug#226570 and #200613

### PR DESCRIPTION
**Pull request by huanwu**
Create the breaking change doc for Bug#226570 and #200613

### CODE REVIEW

File: docs/BreakingChanges/167 WCF message security unable to use TLS1.1 and TLS1.2.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+Edge
+
+### Version Introduced
+4.7
+
+### Source Analyzer Status
+Planned
+
+### Change Description
+.NET 4.7 gave customer the ability to configure either TLS1.1 or TLS1.2 in WCF message security in addition to SSL3.0 and TLS1.0 through the appsettings.
```
Comment (conniey) [inline docs/BreakingChanges/167 WCF message security unable to use TLS1.1 and TLS1.2.md]: Change wording from ".NET 4.7 gave customer the ability to configure either" to "Starting in .NET Framework 4.7, customers can configure either..."

Comment (conniey) [inline docs/BreakingChanges/167 WCF message security unable to use TLS1.1 and TLS1.2.md]: Change wording: "through the appsettings." to "through the Application settings."

File: docs/BreakingChanges/167 WCF message security unable to use TLS1.1 and TLS1.2.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Planned
+
+### Change Description
+.NET 4.7 gave customer the ability to configure either TLS1.1 or TLS1.2 in WCF message security in addition to SSL3.0 and TLS1.0 through the appsettings. 
+
+- [x] Quirked
+- [ ] Build-time break
+
+### Recommended Action
+Prior to 4.7, this change is disabled by default. You can enable this fix by adding the following line to the `<runtime>` section of the app.config or web.config file:
```
Comment (conniey) [inline docs/BreakingChanges/167 WCF message security unable to use TLS1.1 and TLS1.2.md]: Change wording: "Prior to 4.7, this change ..." to "Prior to .NET Framework 4.7, this change ... "

File: docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+Edge
+
+### Version Introduced
+4.7
+
+### Source Analyzer Status
+Planned
+
+### Change Description
+.NET 4.7 fixed the compatibility issue between the escaped strings for some special control characters generated by DataContractJsonSerializer and ECMAScript V6.
```
Comment (conniey) [inline docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md]: Change wording:
".NET 4.7 fixed the compatibility issue between the escaped strings for some special control characters generated by DataContractJsonSerializer and ECMAScript V6."
to
"Starting in .NET 4.7, the compatibility issue between escaped strings for some special control characters generated by DataContractJsonSerializer and ECMAScript V6 has been fixed."

File: docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+Planned
+
+### Change Description
+.NET 4.7 fixed the compatibility issue between the escaped strings for some special control characters generated by DataContractJsonSerializer and ECMAScript V6. 
+
+- [x] Quirked
+- [ ] Build-time break
+
+### Recommended Action
+Prior to 4.7, this change is disabled by default. You can enable this fix by adding the following line to the `<runtime>` section of the app.config or web.config file:
```
Comment (conniey) [inline docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md]: Change wording: "Prior to 4.7, this change ..." to "Prior to .NET 4.7, this change ... "

**Comment by conniey** [comment]
@huanwu Also, your `Affected APIs` are invalid. Our checks reveal 3 invalid docIds:

* `P:System.ServiceModel.Security.TlsnegoTokenAuthenticator.CreateSspiState`
* `P:System.ServiceModel.TlsnegoTokenProvider.CreateTlsSspiState`
* `P:System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject`  (This should be a method with some parameters)

Can you verify in https://apisof.net/ that these docIds exist?

**Comment by huanwu** [comment]
Fixed all issues based on your comments. @conniey

**Comment by conniey** [comment]
@huanwu The doc id for [P:ASystem.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject](https://portability.dot.net/api/hooks/github/pr/docid/details?id=P%3ASystem.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject) is still incorrect.  It is not a property, but a method. Which of those methods does this apply to?

* [M:System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject(System.IO.Stream,System.Object)](https://apisof.net/catalog/System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject(Stream,Object))
* [M:System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject(System.Xml.XmlDictionaryWriter,System.Object)](https://apisof.net/catalog/System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject(XmlDictionaryWriter,Object))
* [M:System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject(System.Xml.XmlWriter,System.Object)](https://apisof.net/catalog/System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject(XmlWriter,Object))

**Comment by huanwu** [comment]
Updated the Affected APIs for all three related method. @conniey

**Comment by conniey** [comment]
@huanwu Those docIds are still wrong. They should be `M:` (method) rather than `P:` (properties).

File: docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md [hunk 5]
```diff
@@ -0,0 +1,35 @@
+## 168: DataContractJsonSerializer incompatible with ES6 (and V8)
```
Comment (rpetrusha) [inline docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md]: This should be expressed in terms of what's fixed, not what's broken. so:
## 168 Serialization of control characters with DataContractJsonSerialzer is now compatible with ES6 (and V8)

File: docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+Edge
+
+### Version Introduced
+4.7
+
+### Source Analyzer Status
+Planned
+
+### Change Description
+Starting in .NET Framework .NET 4.7, the compatibility issue between the escaped strings for some special control characters generated by DataContractJsonSerializer and ECMAScript V6 has been fixed.
```
Comment (rpetrusha) [inline docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md]: I'd change this to:
In the .NET Framework 4.6.2 and earlier versions, the DataContractJsonSerializer did not serialize some special control characters, such as `\b`, `\f`, and `\t`, in a way that was compatible with the ECMAScript V6 and V8 standards. Starting with the .NET Framework 4.7, serialization of these control characters is compatible with ECMAScript V6 and V8.

File: docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+   ```xml
+   <runtime>
+      <AppContextSwitchOverrides value="Switch.System.Runtime.Serialization.DoNotUseECMAScriptV6EscapeControlCharacter=false" />
+   </runtime>
+   ```   
+
+
+### Affected APIs
+* `P:System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject(System.IO.Stream,System.Object)`
```
Comment (rpetrusha) [inline docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md]: `M:System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject(System.IO.Stream,System.Object)`

File: docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+* `P:System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject(System.Xml.XmlDictionaryWriter,System.Object)`
```
Comment (rpetrusha) [inline docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md]: `M:System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject(System.Xml.XmlDictionaryWriter,System.Object)`

File: docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+* `P:System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject(System.Xml.XmlWriter,System.Object)`
```
Comment (rpetrusha) [inline docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md]: `M:System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject(System.Xml.XmlWriter,System.Object)`

File: docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+Planned
+
+### Change Description
+Starting in .NET Framework .NET 4.7, the compatibility issue between the escaped strings for some special control characters generated by DataContractJsonSerializer and ECMAScript V6 has been fixed. 
+
+- [x] Quirked
+- [ ] Build-time break
+
+### Recommended Action
+Prior to .NET 4.7, this change is disabled by default. You can enable this fix by adding the following line to the `<runtime>` section of the app.config or web.config file:
```
Comment (rpetrusha) [inline docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md]: According to DevDiv #378461, this is a retargeting feature. So this should read:
For apps that target the .NET Framework 4.7, this feature is enabled by default. If this behavior is not desirable, you can opt out of this feature by adding the following line to the `<runtime section of the `app.config` or `web.config` file:
```xml
<runtime> 
    <AppContextSwitchOverrides value="Switch.System.Runtime.Serialization.DoNotUseECMAScriptV6EscapeControlCharacter=true" /> 
</runtime>
```
Apps that target the .NET Framework 4.6.2 and earlier versions but are running under the .NET Framework 4.7 or later can opt into this behavior y adding the following line to the `<runtime>` section of the `app.config` or `web.config` file:

File: docs/BreakingChanges/167 WCF message security unable to use TLS1.1 and TLS1.2.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+Edge
+
+### Version Introduced
+4.7
+
+### Source Analyzer Status
+Planned
+
+### Change Description
+Starting in .NET Framework 4.7, customer can configure either TLS1.1 or TLS1.2 in WCF message security in addition to SSL3.0 and TLS1.0 through the Application settings.
```
Comment (mjrousos) [inline docs/BreakingChanges/167 WCF message security unable to use TLS1.1 and TLS1.2.md]: Regarding retargeting/runtime, the 'quirked' box is checked below, so our tools will list this as a retargeting change.

Comment (huanwu) [inline docs/BreakingChanges/167 WCF message security unable to use TLS1.1 and TLS1.2.md]: @rpetrusha, the change is to enable the usage of TLS1.1 or TLS1.2 in WCF service with message security and client credential type Certificate. Before the fix, you are not able to use TLS1.1 or TLS1.2. If SSL3.0 and TLS1.0 are disabled in the registry, the communication will just fail. After the fix, you can specify the protocol you want to use through the following code System.Net.ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls11

File: docs/BreakingChanges/167 WCF message security unable to use TLS1.1 and TLS1.2.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+Planned
+
+### Change Description
+Starting in .NET Framework 4.7, customer can configure either TLS1.1 or TLS1.2 in WCF message security in addition to SSL3.0 and TLS1.0 through the Application settings. 
+
+- [x] Quirked
+- [ ] Build-time break
+
+### Recommended Action
+Prior to .NET Framework 4.7, this change is disabled by default. You can enable this fix by adding the following line to the `<runtime>` section of the app.config or web.config file:
```
Comment (huanwu) [inline docs/BreakingChanges/167 WCF message security unable to use TLS1.1 and TLS1.2.md]: fixed

File: docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md (hunk 5 above)
Comment (huanwu) [inline docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md]: fixed

File: docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md (hunk 6 above)
Comment (huanwu) [inline docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md]: fixed

File: docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md (hunk 10 above)
Comment (huanwu) [inline docs/BreakingChanges/168 DataContractJsonSerializer incompatible with ES6 (and V8).md]: fixed

File: docs/BreakingChanges/167 WCF message security unable to use TLS1.1 and TLS1.2.md [hunk 13]
```diff
@@ -0,0 +1,28 @@
+## 167: WCF message security unable to use TLS1.1 and TLS1.2
```
Comment (mjrousos) [inline docs/BreakingChanges/167 WCF message security unable to use TLS1.1 and TLS1.2.md]: I agree this should probably be phrased the other way.

**Comment by mjrousos** [comment]
These look good to me, Huangli. I had one minor comment about the phrasing of a title (which Ron also commented on), but otherwise they look good.

**Comment by smoro88** [comment]
Has this bug been moved into any release of 4.7 because I am having this exact issue using 4.6.2.

**Comment by dotnetdan** [comment]
Does this fix apply to .NET 4.7 framework, or is it Core only?

**Pull request merged**